drud/ddev
# Only run post-start hook when the new container is actually created, fixes #1115

## The Problem/Issue/Bug:

People who put post-start hooks in really normally want them to run only when the container is created or recreated. A common example is installing an extra Debian package; it only needs to be done when the container is recreated. A start that doesn't recreate doesn't need to run the hook. OP #1115 

## How this PR Solves The Problem:

* Touch /var/tmp/ddev_started.txt on startup; check to see if it's been touched if we're going to do a hook.
* No action is taken if there are no post-start hooks.

## Manual Testing Instructions:

* Create a post-start hook, maybe "bash -c echo hello there"
* `ddev start`. You should see it fire
* `ddev start` again, you should *not* see it fire
* Change the php type and `ddev start`, container should be recreated and you should see it fire.

## Automated Testing Overview:

* Added test for this behavior in TestDdevStart
* Also changed TestDdevStart to only run one CMS, as behavior across all CMSs is the same.

## Related Issue Link(s):
OP #1115 

## Release/Deployment notes:

This does in fact change the behavior of the post-start hook, so needs to be announced. Most people will like it, it's probably more what's expected.
